api/src/test/java/io/opentelemetry/trace/propagation/JaegerPropagatorTest.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.propagation;
+
+import static com.google.common.truth.Truth.assertThat;
+import static io.opentelemetry.trace.propagation.JaegerPropagator.PARENT_SPAN;
+import static io.opentelemetry.trace.propagation.JaegerPropagator.SEPARATOR;
+import static io.opentelemetry.trace.propagation.JaegerPropagator.TRACE_ID_HEADER;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.internal.StringUtils;
+import io.opentelemetry.trace.DefaultSpan;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.SpanId;
+import io.opentelemetry.trace.TraceFlags;
+import io.opentelemetry.trace.TraceId;
+import io.opentelemetry.trace.TraceState;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link io.opentelemetry.trace.propagation.JaegerPropagator}. */
+@RunWith(JUnit4.class)
+public class JaegerPropagatorTest {
```

Comment:
Tests could use Jaeger client and compare results on in the carrier with this implementation.

Reply:
Thanks - I've added the Jaeger client and used the  ```TextMapCodec``` to generate the value to compare against.